common-util/api/predictionRequests.js

```diff
@@ -9,24 +11,47 @@ export const getPredictionRequests = async () => {
     },
   });
 
-  const markets = response.data.all_markets;
-  const prefilteredRequests = Object.keys(markets).map((key) => ({
-    ...markets[key],
-    key: markets[key].id,
+  const requests = response.data.all_markets;
+  const prefilteredRequests = Object.keys(requests).map((key) => ({
+    ...requests[key],
+    key: requests[key].id,
   }));
 
-  const requests = prefilteredRequests.filter(
+  const allRequests = prefilteredRequests.filter(
     (market) => market.source === 'contribute',
   );
 
-  return requests;
+  const processedRequests = allRequests.filter(
+    (market) => market.state === 'PROCESSED',
+  );
+
+  const approvedRequestsCount = allRequests.filter(
+    (market) => market.state === 'APPROVED',
+  ).length;
+
+  return {
+    allRequests,
+    approvedRequestsCount,
+    processedRequests,
+  };
 };
 
 export const postPredictionRequest = async (payload) => {
-  await axios.post(PREDICT_BASE_URL + PREDICT_REQUEST_ENDPOINT, payload, {
-    headers: {
-      Authorization: process.env.NEXT_PUBLIC_PREDICT_API_KEY,
-      'Content-Type': 'application/json',
-    },
-  });
+  const headers = {
+    Authorization: process.env.NEXT_PUBLIC_PREDICT_API_KEY,
+    'Content-Type': 'application/json',
+  };
+
+  const response = await axios.post(
+    PREDICT_BASE_URL + PREDICT_PROPOSE_ENDPOINT,
+    payload,
+    { headers },
+  );
+
+  // Extract the id from the response data
+  // Respond structure data.info = "Market ID {id} created successfully."
+  const id = response.data.info.split(' ')[2];
```

Comment:
This is a very brittle way of getting the id. Let's keep as-is for now, but you should ask an output JSON wiht an approppriate output, e.g.:
{ "info": "message", "id": "00000"} etc.

Reply:
Yes that's ideal – just went with what was available for now